src/main/java/org/springframework/data/elasticsearch/core/DefaultResultMapper.java

```diff
@@ -105,13 +96,7 @@ private static EntityMapper initEntityMapper(
 		List<T> results = new ArrayList<>();
 		for (SearchHit hit : response.getHits()) {
 			if (hit != null) {
-				T result = null;
-				String hitSourceAsString = hit.getSourceAsString();
-				if (!StringUtils.isEmpty(hitSourceAsString)) {
-					result = mapEntity(hitSourceAsString, clazz);
-				} else {
-					result = mapEntity(hit.getFields().values(), clazz);
-				}
+				T result = mapSearchHit(hit, clazz);
```

Comment:
what if `mapSearchHit`returns `null`?

Reply:
Why should it be null when we have a hit? We can probably get an uninitialized object if fields are empty.

Reply:
`mapSearchHit` returns `@Nullable` that's why I would have a check here.